### DISCUSSION TITLE
How to use datatype.struct?

### DISCUSSION THREAD

**Discussion by isavygin** [Koenkk/zigbee2mqtt#23875] [Devices]
I apologize if it is not a proper place for that kind of discussion.
I'm trying to add new my custom device to zigbee2mqtt. Device has an atrribute of struct type. As I understood I should use writeStructure method for that. I've managed to send an array, but I have problems how to form payload is datatype is struct.
```
                const payload_str = [];
                payload_str.push({elmType: enums_1.DataType.UINT8, elmVal: 3});
                payload_str.push({elmType: enums_1.DataType.UINT16, elmVal: 4});        
                var payload = [{
                        attrId : 61444,
                        dataType : enums_1.DataType.STRUCT,
                        selector : {},
                        elmData : payload_str,
                }];
                await entity.writeStructured('manuCustomDevice', payload);
```
But in the end I have the error in the log: 
Cannot read properties of undefined (reading 'length')
Could anyone provide me an example how to send data of STRUCT type?

**Comment by ghost** [comment]
Hello,

Did you find any solution for this?

I've got the same issue with a `DataType.ARRAY` type.
